client/core/trade.go

```diff
@@ -746,6 +759,12 @@ func (t *trackedTrade) swapMatches(matches []*matchTracker) error {
 		}
 	}
 
+	// Check if we need to return the change coin because the order has been
+	// canceled.
+	if t.metaData.Status == order.OrderStatusCanceled && !t.hasUnswappedMatches() {
+		t.wallets.fromWallet.ReturnCoins(asset.Coins{t.change})
+	}
+
```

Comment:
The TODO comment on line 747 (`SetChangeCoin`) can go now that `Swap` is doing this.